org.eclipse.jdt.debug.ui/ui/org/eclipse/jdt/internal/debug/ui/console/JavaStackTraceHyperlink.java

```diff
@@ -203,6 +252,96 @@ public IStatus runInUIThread(IProgressMonitor monitor) {
 		job.schedule();
 	}
 
+	/**
+	 * Filters results based on method signature
+	 *
+	 * @param type
+	 *            Type of the matches
+	 * @param methodSignature
+	 *            entire method declaration
+	 * @param methodNameExtracted
+	 *            method name
+	 * @throws JavaModelException
+	 */
+	private boolean extractFromResults(IType type, String methodSignature, String methodNameExtracted) throws JavaModelException {
+		IMethod[] methods = type.getMethods();
+		for (IMethod method : methods) {
+			int indexOfClosing = method.toString().indexOf(')');
+			int indexOfStart = method.toString().indexOf(method.getElementName());
+			String methodName = method.toString().substring(indexOfStart, indexOfClosing + 1).replaceAll(" ", ""); //$NON-NLS-1$//$NON-NLS-2$
+			int paramCount = methodSignature.substring(methodSignature.indexOf('(')
+					+ 1, methodSignature.lastIndexOf(')')).split(",").length; //$NON-NLS-1$
+			if (methodName.equals(methodSignature)) {
+				return true;
+			} else if (methodNameExtracted.equals(method.getElementName())
+					&& paramCount == method.getNumberOfParameters()) {
+				// Further mining from fully qualified parameter names in method signature
+				StringBuilder methodSignatureBuilder = new StringBuilder(method.getElementName());
+				methodSignatureBuilder.append('(');
+				String[] params = methodName.split(","); //$NON-NLS-1$
+				for (String block : params) {
+					if (block.contains("...")) { //$NON-NLS-1$ Parameter is var args
```

Comment:
Code inside `if (block.contains("..."))` handles some specific case - let refactor it into a dedicated method

Reply:
Yes it is for handling variable arguments, now refactored to a dedicated method